### PR TITLE
Problem: fixes in ibc-go v5.1 are not included

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- make a breaking change to upgrade to ibc-go `v5.1.0`.
- make the v1.0.0 upgrade works for both testnet and mainnet.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Changelog
 
-*Nov 9, 2022*
+## UNRELEASED
+
+### State Machine Breaking
+
+- [#]() Upgrade ibc-go to [v5.1.0](https://github.com/cosmos/ibc-go/releases/tag/v5.1.0).
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
- [#765](https://github.com/crypto-org-chain/cronos/pull/765) Upgrade ibc-go to [v5.1.0](https://github.com/cosmos/ibc-go/releases/tag/v5.1.0).
```

File: go.mod [hunk 2]
```diff
@@ -191,8 +191,8 @@ require (
 replace (
 	// ics23 patch for dragonberry
 	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221109093801-2899dcf82f16
-	github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.13-0.20221108065429-afacc02b8738
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221109202200-2bd2cf1f3aae
+	github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.19-deepcopy-jumptable
```
Comment (yihuang) [inline go.mod:195]: re-cherrypick the optimized solution eventually merged into go-ethereum main branch.

File: go.mod [hunk 3]
```diff
@@ -191,8 +191,8 @@ require (
 replace (
 	// ics23 patch for dragonberry
 	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221109093801-2899dcf82f16
-	github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.13-0.20221108065429-afacc02b8738
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221109202200-2bd2cf1f3aae
```
Comment (yihuang) [inline go.mod:194]: can wait a little bit for cosmos-sdk 0.46.5 release

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # Changelog
 
-*Nov 9, 2022*
+## UNRELEASED
```
Comment (yihuang) [inline CHANGELOG.md:3]:
```suggestion
## v1.0.0-rc3
```

File: go.mod [hunk 5]
```diff
@@ -191,8 +191,8 @@ require (
 replace (
 	// ics23 patch for dragonberry
 	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221109093801-2899dcf82f16
-	github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.13-0.20221108065429-afacc02b8738
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.5-0.20221111074045-68e54fa5b868
```
Comment (yihuang) [inline go.mod:194]: https://github.com/cosmos/cosmos-sdk/compare/2899dcf82f16...68e54fa5b868

File: go.mod [hunk 6]
```diff
@@ -202,6 +202,7 @@ replace (
 
 	// TODO: remove when gravity update dependencies
 	github.com/peggyjv/gravity-bridge/module/v2 => github.com/crypto-org-chain/gravity-bridge/module/v2 v2.0.1-0.20220815102151-48275db7e1ee
+	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.24-0.20221110131553-ec471ba27efd
```
Comment (yihuang) [inline go.mod:205]: 0.34.x branch, include the check_tx cache fix.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,12 @@
 # Changelog
 
-*Nov 9, 2022*
+## v1.0.0-rc3
+
+### State Machine Breaking
+
+- [#765](https://github.com/crypto-org-chain/cronos/pull/765) Upgrade ibc-go to [v5.1.0](https://github.com/cosmos/ibc-go/releases/tag/v5.1.0).
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
- [#765](https://github.com/crypto-org-chain/cronos/pull/765) Upgrade ibc-go to [v5.1.0](https://github.com/cosmos/ibc-go/releases/tag/v5.1.0) and related dependencies.
```

File: go.mod (hunk 6 above)
Comment (mmsqe) [reply go.mod:205]: thanks jay [!](https://github.com/tendermint/tendermint/pull/9650/files)

**Review by mmsqe — COMMENTED** [review]
do we need a test for the send enabled later?

File: cmd/cronosd/config/config.go [hunk 8]
```diff
@@ -2,6 +2,11 @@ package config
 
 import sdk "github.com/cosmos/cosmos-sdk/types"
 
+const (
+	NetworkMainet = iota + 1
```
Comment (JayT106) [inline cmd/cronosd/config/config.go:6]: suggest change:  NetworkMainnet

Comment (JayT106) [reply cmd/cronosd/config/config.go:6]: update: maybe just call `Mainnet`?

File: cmd/cronosd/config/config.go [hunk 9]
```diff
@@ -2,6 +2,11 @@ package config
...
+	NetworkTestnet
```
Comment (JayT106) [inline cmd/cronosd/config/config.go:7]: Maybe consider `TestnetPoseidon` or `Testnet<XXX>`? if we will have more than 1 testnet in the future?

File: go.mod (hunk 6 above)
Comment (JayT106) [reply go.mod:205]: thanks @yihuang

**Review by JayT106 — COMMENTED** [review]
LGTM, just leave some comments.

File: cmd/cronosd/config/config.go (hunk 9 above)
Comment (yihuang) [reply cmd/cronosd/config/config.go:7]: yeah, I guess it's not a good idea to do the check here, we don't know about different testnet networks at compile time, so I added a different plan name for testnet3 to upgrade, `v1.0.0-testnet3`.

File: cmd/cronosd/config/config.go (hunk 8 above)
Comment (yihuang) [reply cmd/cronosd/config/config.go:6]: removed, use a different upgrade plan name for testnet3.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**